### ISSUE TITLE
Reduce scaffolding time by introducing a helper

### ISSUE THREAD

**Issue by ash123456789** [langleyfoxall/react-dynamic-data-table#7]
After using both this package and the [Laravel Datatables](https://github.com/yajra/laravel-datatables) package I realised that this package could potentially be improved by mimicking what the datatables package does.

The datatables package has its own helper which then wraps around `Model::query()` and performs all its filtering and pagination on that singular endpoint. This would be ideal to improve the set up time rather than having to repeat yourself when creating the individual endpoints for filtering, searching, etc.

### How datatables does it
In your method you can use the following so that the package will then perform all its requests against a singular endpoint.

```php
return datatables(PaymentMethod::query())->toJson();
```
The helper is just the following:
```php
    /**
     * Helper to make a new DataTable instance from source.
     * Or return the factory if source is not set.
     *
     * @param mixed $source
     * @return \Yajra\DataTables\DataTableAbstract|\Yajra\DataTables\DataTables
     */
    function datatables($source = null)
    {
        if (is_null($source)) {
            return app('datatables');
        }

        return app('datatables')->make($source);
    }
```

### What we could do

We could have our own helper which would perform largely the same and improve scaffolding time. Something along the lines of...

```php
return reactdatatables(PaymentMethod::query())->toJson();
```

**Comment by DivineOmega** [comment]
@ash123456789 It's a very good idea. The JS side of this could be implemented either as a separate package that wraps around this one, or as an `apiUrl` prop that overrides the behaviour usually specified by other props.

**Comment by DivineOmega** [comment]
So, we could do something like the following JS side:

```jsx
<DynamicDataTable apiUrl='http://example.com/api/users' />
```

Then PHP side, this URL could be handled by controller provided by a separate `laravel-react-dynamic-data-table` package or similar.

**Comment by DivineOmega** [comment]
I'll work on this. It should really improve the speed as which we can do basic setup for listing pages.

**Comment by DivineOmega** [comment]
I have started work on the Laravel API side of this issue in a seperate repo.

https://github.com/langleyfoxall/react-dynamic-data-table-laravel-api

**Comment by DivineOmega** [comment]
I've started work on the React side of this. It seems like is probably going to work better if I create a wrapper component that will work similar to the following.

```jsx
<DynamicDataTable apiUrl='http://example.com/api/users' />
```

**Comment by AlexCatch** [comment]
@DivineOmega I've done pagination in React more times than I can count so I abstracted the logic out into it's own class `PaginatedDataSource` that encapsulates all the logic, I can share it with you if you want?

**Comment by DivineOmega** [comment]
@AlexCatch I noticed that in the Elm users section when I was using it as reference for the implemented. Looked very well abstracted.

To get it done quickly, I am implementing it in the most obvious manner first, but feel free to PR in your abstraction when I'm done. :+1:

**Comment by DivineOmega** [comment]
The React side of this is now available in this package, in version 1.0.9. The usage syntax is as follows.

```jsx
<AjaxDynamicDataTable apiUrl={'http://example.test/web-api/users'}/>
```

The endpoint can be automatically provided by the [React Dynamic Data Table Laravel API](https://github.com/langleyfoxall/react-dynamic-data-table-laravel-api) package. This still requires some final testing, before its initial release.

**Comment by DivineOmega** [comment]
With the v1.0.0 release of the Laravel API for React Dynamic Data Table I'm considering this complete. 🙂 

https://github.com/langleyfoxall/react-dynamic-data-table-laravel-api
